libraries/redcore/api/soap/operation/operation.php

```diff
@@ -125,51 +121,71 @@ public function readItem($id = array(), $language = null)
 			$primaryKeysFromFields['id'] = array('transform' => 'int');
 		}
 
-		if (!is_array($id))
+		foreach ($primaryKeysFromFields as $primaryKey => $primaryKeyField)
 		{
-			if (count($primaryKeysFromFields) == 1)
-			{
-				$idValue = $id;
-				$id = array();
-				$id[key($primaryKeysFromFields)] = $idValue;
-			}
-			else
+			$keyData = '';
+
+			if (isset($data->$primaryKey) && $data->$primaryKey != '')
 			{
-				$id = array('id' => $id);
+				$keyData = $data->$primaryKey;
 			}
+
+			$dataGet->$primaryKey = $this->webservice->transformField($primaryKeyField['transform'], $keyData, false);
 		}
 
+		// Handle different language switch
+		$this->setLanguage((string) (isset($data->language) ? $data->language : ''));
+
+		$this->webservice->options->set('dataGet', $dataGet);
+		$this->webservice->options->set('task', '');
+		$this->webservice->options->set('filterOutResourcesGroups', array('_links', '_messages'));
+		$this->webservice->execute();
+
+		$arr = $this->webservice->hal->toArray();
+		$outputResources = RApiSoapHelper::getOutputResources($this->webservice->configuration->operations->read->item, '', true);
+
+		$response = RApiSoapHelper::selectListResources($outputResources, array($arr));
+
+		$final = new stdClass;
+		$final->item = $response[0];
+
+		$match = true;
+
 		foreach ($primaryKeysFromFields as $primaryKey => $primaryKeyField)
 		{
-			if (isset($id[$primaryKey]) && $id[$primaryKey] != '')
+			if ($dataGet->$primaryKey != $final->item->$primaryKey)
```

Comment:
why is this check here? you should never receive item with different primary key...